packages/xs-vat-worker/package.json

```diff
@@ -0,0 +1,57 @@
+{
+  "name": "@agoric/xs-vat-worker",
+  "version": "0.1.0",
+  "description": "swingset-vat worker for XS js runtime",
+  "main": "src/locate.js",
+  "module": "src/locate.js",
+  "scripts": {
+    "postinstall": "cp src/locate-undefined.js src/locate.js",
```

Comment:
I'm a little hesitant to lean on `postinstall` or to have some generated file with an absolute path in it, in case someone moves their source tree. What's wrong with putting a `require.resolve()` and an `fs.exists` check in `src/locate.js`? Are you avoiding `require.resolve` because it doesn't exist in ESM and we'll have to rewrite it in terms of some sort of `import.meta.url` construction?

Reply:
What I'm avoiding is exporting functions with ambient authority. i.e. I'm sticking to ocap discipline when it comes to imported modules.

I thought about having the caller pass in `require.resolve` and `fs.exists` but that seemed a little complex. `require.resolve` has a sort of link-time feel, but it's really a runtime thing and I've been burned by treating it otherwise. Also, the info about where the binary ends up is in the `xs-lin.mk` makefile (which gets it from the moddable SDK) so it seemed reasonable to treat it as a build-time thing. But I'm not wedded to this approach. Mostly I'm glad you're looking closely enough to notice it :)

brainstorming... treat `require.resolve` as link-time thing (i.e. exempt from ocap discipline) for coming up with the pathname, but then let the caller do the exists check? Clearly for the path to be useful to the caller, the caller has to have `spawn` power which pretty much subsumes `fs.exists` power.

more brainstorming... I gather endo learned about JSON modules recently... what are the odds it will learn to import binary blobs? hmm... that's a stretch.

Also... now that I think about it... why return undefined rather than throwing? Surely that invites the risk that the failure will quietly propagate far from where it originated, making it difficult to diagnose?

Reply:
ah, ok, thanks for reminding me of the motivation

`require.resolve` is different for each module (since it embeds the referrer's location), so it's not something that could be usefully passed *into* a different module: the pathnames would be all wrong, and the whole point of this function is to isolate the `packages/xs-vat-worker/` directory layout from the caller.

I'd be ok with `xs-vat-worker` exporting something that returns pathname-or-undefined, and having the caller do the `exists` check.

I wouldn't want that something to throw, because half the point is to detect whether or not the XS worker was compiled or not. At this stage, "not compiled" is a normal case, not an exception. If/when we get to the point where it's only missing if some drastic tooling infrastructure failed, then yeah in that case we could make it raise an exception.

Reply:
TIL: `require.resolve` expects to find javascript code. It raises MODULE_NOT_FOUND if it can only find an executable there.

Reply:
I propose:

```js
/**
 * Locate the XS vat worker executable.
 *
 * Note: executable is not built by default.
 * @see the `build:xs-lin` script in package.json
 *
 * @param {{ resolve: (...string) => string }} filesystem path access
 * @returns { string } full path where linux debug executable is built;
 *                     not guaranteed to exist.
 */
export function locateWorkerBin({ resolve }) {
```

(I'd give the git hash of my proposal but master moves so fast I have to rebase quite frequently; and I've been squashing more lately.)